pyproject.toml

```diff
@@ -48,7 +47,7 @@ include = [
 zenml = "zenml.cli.cli:cli"
 
 [tool.poetry.dependencies]
-python = ">=3.6.2,<3.9"
+python = ">=3.7,<3.9"
```

Comment:
My suggestion would be to directly set the minimum as 3.7.1 then we can use pandas 1.3 instead of 1.1 — we're pretty far behind there

Reply:
Fine with me, let's see if @htahir1 has any objections?

Reply:
Worth checking mixpanel, too, to see if we have any people confirmed as using those versions. Last time I checked we had no 3.6 users any more.

Reply:
Seems like no one is using anything below python 3.7.5 in mixpanel actually

Reply:
3.7.1 should be fine

Reply:
for what it's worth, 3.7.1 is older than things like 3.4.10 or 3.6.8, so it's really quite unlikely to be in use

Reply:
Changed it to >= 3.7.1